src/item.cpp

```diff
@@ -8203,6 +8205,33 @@ cata::optional<tripoint> item::get_cable_target( player *p, const tripoint &pos
 
 bool item::process_cable( player *p, const tripoint &pos )
 {
+    if( p == nullptr ) {
+        reset_cable( p );
+    }
+    std::string state = get_var( "state" );
+    if( state == "solar_pack_link" ) {
+        if( !p->has_item( *this ) || ( !p->is_wearing( "solarpack_on" ) ||
+                                       !p->is_wearing( "solarpack_on" ) ) ) {
+            p->add_msg_if_player( m_bad, _( "You notice the cable has come loose!" ) );
+            reset_cable( p );
+            return false;
+        }
+    }
+
+    static const item_filter used_ups = [&]( const item & itm ) {
+                    return itm.get_var( "cable" ) == "plugged_in";
+                    };
+
+    if( state == "UPS" ) {
+        if( !p->has_item( *this ) || ( !p->has_item_with( used_ups ) ) ) {
```

Comment:
```suggestion
        if( !p->has_item( *this ) || !p->has_item_with( used_ups ) ) {
```
And needs a check for `p` not being null.

Reply:
As above.